### PR TITLE
SUBMARINE-66. Improve TF config env JSON generator + tests

### PR DESCRIPTION
**Pull request by adamantal**
### What is this PR for?
* The goal is to update the TensorFlow config generation. Also some of the tests were doing some manual JSON thing which is not very maintainable - it is also changed to a more flexible one using jackson-databind

### What type of PR is it?
* Refactoring | Test

### Todos
* [x] - Travis checks
* [x] - New tests should pass

### What is the Jira issue?
* [SUBMARINE-66](https://issues.apache.org/jira/browse/SUBMARINE-66)

### How should this be tested?
* Only the new UTs should pass.

### Questions:
* Does the licenses files need update? Yes
  * We should probably add jackson-module-jaxb-annotations?
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGenerator.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      ArrayNode array = cluster.putArray(name);
+      for (int i = 0; i < count; i++) {
+        String componentValue = String.format("%s-%d%s", name, i,
+            endpointSuffix);
+        array.add(componentValue);
+      }
+    }
+  }
+
+
```
Comment (jiwq) [inline submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGenerator.java]: redundant blank line

Comment (adamantal) [reply submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGenerator.java]: Thanks for the review! Removed the blank line.

**Comment by adamantal** [comment]
Rebase & force push since one of Travis check failed due to some external reason.

**Comment by adamantal** [comment]
Seems like all the checks have passed. Do you have any additional comment @jiwq ?

File: submarine-server/server-submitter/submitter-yarnservice/src/test/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Class to test some functionality of {@link TensorFlowConfigEnvGenerator}.
+ */
+public class TensorFlowConfigEnvGeneratorTest {
+  private ObjectMapper objectMapper;
+
+  @Before
+  public void setUp() {
```
Comment (jiwq) [inline submarine-server/server-submitter/submitter-yarnservice/src/test/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGeneratorTest.java]: `setup` maybe better than `setUp`

Comment (adamantal) [reply submarine-server/server-submitter/submitter-yarnservice/src/test/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGeneratorTest.java]: Done. Thanks for the review!

**Comment by liuxunorg** [comment]
@adamantal , Thank you contribute.

**Comment by yuanzac** [comment]
Thanks, @adamantal for your contributions and @jiwq for the review

**Comment by adamantal** [comment]
Thanks for the review and the commits!

**Pull request closed without merging**